src/VisualStudio/VisualBasic/Impl/Debugging/LocationInfoGetter.vb

```diff
@@ -3,58 +3,117 @@
-                ' model below then it might take a *long* time as all dependent compilations are built.
-
+                ' PERF:  This method will be called synchronously on the UI thread for every breakpoint in the solution.
+                ' Therefore, it is important that we make this call as cheap as possible.  Rather than constructing a
+                ' containing Symbol and using ToDisplayString (which might be more *correct*), we'll just do the best we
+                ' can with Syntax.  This approach is capable of providing parity with the pre-Roslyn implementation.
                 Dim tree = Await document.GetVisualBasicSyntaxTreeAsync(cancellationToken).ConfigureAwait(False)
                 Dim root = Await tree.GetRootAsync(cancellationToken).ConfigureAwait(False)
                 Dim token = root.FindToken(position)
-                Dim memberDecl = token.GetContainingMemberBlockBegin()
+                Dim memberDeclaration = token.GetContainingMember()
+                ' Unlike C#, VB doesn't show field names.
+                If memberDeclaration.Kind = SyntaxKind.FieldDeclaration Then
+                    memberDeclaration = memberDeclaration.GetAncestor(Of DeclarationStatementSyntax)()
+                End If
+
+                If memberDeclaration Is Nothing Then
+                    Return Nothing
+                End If
+
+                Dim compilation = Await document.GetVisualBasicCompilationAsync(cancellationToken).ConfigureAwait(False)
+                Dim name = GetName(memberDeclaration, compilation.Options.RootNamespace)
+
+                Dim text = Await document.GetTextAsync(cancellationToken).ConfigureAwait(False)
+                Dim lineNumber = text.Lines.GetLineFromPosition(position).LineNumber
+                Dim memberLine = text.Lines.GetLineFromPosition(memberDeclaration.GetMemberBlockBegin().SpanStart).LineNumber
+                Dim lineOffset = lineNumber - memberLine
+
+                Return New DebugLocationInfo(name, lineOffset)
+            End Function
+
+            Private Function GetName(memberDeclaration As DeclarationStatementSyntax, rootNamespace As String) As String
+                Const missingInformationPlaceholder As String = "?"
+                Const dotToken As Char = "."c
+
+                ' root namespace
+                Dim pooled = PooledStringBuilder.GetInstance()
+                Dim builder = pooled.Builder
+                If Not String.IsNullOrEmpty(rootNamespace) Then
+                    builder.Append(rootNamespace)
+                    builder.Append(dotToken)
+                End If
+                ' containing namespace(s) and type(s)
+                Dim containingDeclarationNames As ArrayBuilder(Of String) = ArrayBuilder(Of String).GetInstance()
+                Dim containingDeclaration = memberDeclaration.Parent
+                If TypeOf memberDeclaration Is TypeStatementSyntax Then
```

Comment:
It's a special case in VB if you manage to get a breakpoint set at a span that is in a class (but not in one of the members of the class).  In that case, we want to skip the class in the loop below (and just append the name of the class where we do ```' simple name```.